### DISCUSSION TITLE
Brew tap fails if `binary` format for archives are used.

### DISCUSSION THREAD

**Discussion by fogfish** [goreleaser/goreleaser#3843] [Ideas, Issue Triage, and General discussions]
If project uses the binary archives

```
archives:
  - format: binary
```

it fails to install brewtap
```
Errno::ENOENT: No such file or directory - ...
```

The rootcause seems to be `bin.install`. The brew downloads the binary file but `bin.install` assumes tarball

```
on_macos do
    if Hardware::CPU.intel?
      url "https://github.com/abc/abc/releases/download/v1.0.5/abc_1.0.5_darwin_amd64"
      sha256 "883eea2c809bae2b6d49b7420f1349c9066f2bd4bbc993806e0586a3fad7efe0"

      def install
        bin.install "abc"
      end
    end
   end
``` 

Is this a know limitations, bug or user error?

Thank you in advanced.

**Comment by caarlos0** [comment]
that seems like a regression... I worked on it years ago! will open an issue for it

thanks for reporting!

**Reply by caarlos0** [reply]
lets continue in #3844
